BUG: Fix update notice when idc-index-data has a new version

Sunderlandkyl opened the pull request:
idc-index-data would be marked as out of date, however it would not be upgraded, making it impossible to get rid of the update required message. Fixed by calling --upgrade on idx-index-data to trigger the update.

Also reload the idc-index library after updating so that the new library is used.

fedorov commented:
Oh wow! But I am surprised - I was pretty sure when idc-index is upgraded, it will pull in upgraded idc-index-data, and I am pretty sure it worked for me that way.

In any case, this seems to be harmless, so I definitely can merge and we can discuss when we meet tomorrow. Or let me know and I can wait until we discuss tomorrow.

fedorov commented on IDCBrowser/IDCBrowser.py:
```diff
@@ -1521,8 +1522,26 @@ def setupPythonRequirements(self, update=False):
+          slicer.util.pip_install("--no-cache-dir --upgrade idc-index>=0.7.0")
+          slicer.util.pip_install("--no-cache-dir --upgrade idc-index-data")
+        else:
+          slicer.util.pip_install("idc-index>=0.7.0 idc-index-data")
```
@Sunderlandkyl I am thinking we can remove that version constraint altogether. I would expect upgrade to pull the latest version, so that outdated 0.7.0 only adds confusion - there's nothing special about that version really.

Sunderlandkyl replied:
Ok, I've removed the version number.

Sunderlandkyl commented:
As far as I can tell the --upgrade argument only applies to the first library, so "idc-index-data" wasn't being updated.
Should be good to go now.

fedorov commented:
> As far as I can tell the --upgrade argument only applies to the first library, so "idc-index-data" wasn't being updated.

Right, but the dependency for idc-index is always updated to specify new version of idc-index-data. So for example the latest idc-index specifically requests idc-index-data 23.0.1: https://github.com/ImagingDataCommons/idc-index/blob/main/pyproject.toml#L35

Sunderlandkyl commented:
I thought this PR had been integrated, but it seems I had forgotten about it. I've updated it so that we only check idc-index and ignore idc-index-data updates.

I've also added some changes to improve the stability of the update checking/installation processes.

Sunderlandkyl committed:
BUG: Fix update notice when idc-index-data has a new version

idc-index-data would be marked as out of date, however it would not be upgraded, making it impossible to get rid of the update required message.
Fixed by only checking if idc-index is out of date.

Also, improved the update process, so that the module is updated before restart once the button is clicked.

Re #76

fedorov commented:
@Sunderlandkyl please merge when you consider this is ready.

Sunderlandkyl commented:
@fedorov Ok, I've gone ahead and merged the changes.
